### PR TITLE
feat(hydroflow): Added poll_futures and poll_futures_async operators.

### PR DESCRIPTION
**Pull request by RyanAlameddine**

### CODE REVIEW

File: hydroflow/src/util/mod.rs [hunk 1]
```diff
@@ -110,10 +110,13 @@ where
...
-    let any = std::cell::Cell::new(true);
-    let mut unfused_iter = ready_iter(stream).inspect(|_| any.set(true));
+    use std::sync::atomic::Ordering;
+
+    let got_any_items = std::sync::atomic::AtomicBool::new(true);
+    let mut unfused_iter =
+        ready_iter(stream).inspect(|_| got_any_items.store(true, Ordering::Relaxed));
     let mut out = C::default();
-    while any.replace(false) {
+    while got_any_items.swap(false, Ordering::Relaxed) {
```
Comment (MingweiSamuel) [inline hydroflow/src/util/mod.rs:119]: This doesn't seem to change anything? Does it make the future `Send` or something?

File: benches/benches/futures.rs [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+        let done = Rc::new(RefCell::new(false));
+        let mut wakers = Vec::new();
+
+        let range = 0..count;
+        let futs = range
+            .map(|i| {
+                let waker = Rc::new(RefCell::new(None));
+                let d = if !wake_one || i == 0 {
+                    wakers.push(waker.clone());
+                    done.clone()
```
Comment (MingweiSamuel) [inline benches/benches/futures.rs:77]: Oh nevermind, I see here

File: hydroflow/src/util/mod.rs (hunk 1 above)
Comment (MingweiSamuel) [reply hydroflow/src/util/mod.rs:119]: Yup needs to be `Sync` for the returned future to be `Send`

**Comment by shadaj** [comment]
One thing I thought of today... does the availability of new future results automatically schedule the subgraph? As in if there are no new inputs but new resolved futures does the code still run.

File: hydroflow_lang/src/graph/ops/poll_futures.rs [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    let write_prologue = quote_spanned! {op_span=>
+        let #futures_ident = df.add_state(
+            ::std::cell::RefCell::new(
+                #root::futures::stream::#future_type::new()
+            )
+        );
+    };
+
+    let write_iterator = if is_pull {
+    let input = &inputs[0];
```
Comment (shadaj) [inline hydroflow_lang/src/graph/ops/poll_futures.rs:58]: This indentation is funky

**Comment by RyanAlameddine** [comment]
> One thing I thought of today... does the availability of new future results automatically schedule the subgraph? As in if there are no new inputs but new resolved futures does the code still run.

Yes. A waker which schedules the subgraph is passed into the poll calls.

This is tested in the multi-batch tests as well.

**Review by shadaj — APPROVED** [review]
Modulo weird formatting, LGTM!

**Comment by RyanAlameddine** [comment]
Added Hydroflow+ operators, re-requesting review for final commit

**Review by shadaj — APPROVED** [review]

**Pull request merged**